src/sinks/nats.rs

```diff
@@ -86,10 +88,14 @@ impl SinkConfig for NatsSinkConfig {
 
 impl NatsSinkConfig {
     fn to_nats_options(&self) -> async_nats::Options {
-        // Set reconnect_buffer_size on the nats client to 0 bytes so that the
-        // client doesn't buffer internally (to avoid message loss).
-        async_nats::Options::new()
+        let options = match &self.auth {
+            None => async_nats::Options::new(),
+            Some(auth) => auth.to_nats_options(),
+        };
```

Comment:
Consider using `map` for this:
```suggestion
        let options = self.auth.as_ref()
            .map(|auth| auth.to_nats_options())
            .unwrap_or_else(|| async_nats::Options::new());
```
(`rustfmt` will undoubtedly reformat that line)